src/main/host/process.rs

```diff
@@ -467,15 +467,8 @@ impl RunnableProcess {
         let prev_total = self.total_run_time.replace(total_elapsed);
         let delta = total_elapsed - prev_total;
 
-        if let Some(mut tracker) = host.tracker_borrow_mut() {
-            unsafe {
-                cshadow::tracker_addProcessingTimeNanos(
-                    &mut *tracker,
-                    delta.as_nanos().try_into().unwrap(),
-                )
-            };
-            host.cpu_borrow_mut().add_delay(delta);
-        }
+        host.cpu_borrow_mut().add_delay(delta);
```

Comment:
Was this a bug before, that we were only adding the delay if the tracker was enabled?

Actually is this still a bug, that we only add the cpu delay if perf timers are enabled? I don't think we want enabling performance timers to change the simulation behaviour?

I don't actually know how this cpu object is supposed to work, but it's used in `Host::execute` to reschedule events.

Reply:
Yes, I think this was a bug, and we probably want to add even if perf timers are not enabled. I don't think the CPU module does anything with those delays now, but still would be nice to call the `add_delay()` function in the correct spot :)

Reply:
The CPU was designed to accumulate delay and then if a delay threshold is exceeded, we would simulate a "blocked" CPU by rescheduling events into the future at a time when the CPU became "unblocked". However, I believe this CPU blocking feature got disabled when rewriting to Rust, because we didn't understand it well and the current design potentially causes nondeterministic behavior.

It is configured as part of the `HostInfo` structure but is initialized to `None` here:

https://github.com/shadow/shadow/blob/d2a8eb7302b65451ac66a17cf39549dba137e8e1/src/main/core/sim_config.rs#L249-L250

Which gets copied by the manager for each host in the simulation:

https://github.com/shadow/shadow/blob/d2a8eb7302b65451ac66a17cf39549dba137e8e1/src/main/core/manager.rs#L577-L578

And passed into `Cpu::new` on host creation:

https://github.com/shadow/shadow/blob/d2a8eb7302b65451ac66a17cf39549dba137e8e1/src/main/host/host.rs#L234-L239

So the CPU threshold is `None` and the CPU always returns that it has zero delay:

https://github.com/shadow/shadow/blob/d2a8eb7302b65451ac66a17cf39549dba137e8e1/src/main/host/cpu.rs#L81-L94

and thus we never execute the event rescheduling block in here:

https://github.com/shadow/shadow/blob/d2a8eb7302b65451ac66a17cf39549dba137e8e1/src/main/host/host.rs#L820-L848

Reply:
Tracking native CPU performance times is performance sensitive, which is probably why `host.cpu_borrow_mut().add_delay(delta);` is configured behind the `perf_timers` feature. I'm not going to reverse that decision in the PR.